uaclient/jobs/gcp_auto_attach.py

```diff
@@ -9,24 +9,27 @@
 
 
 def gcp_auto_attach(cfg: config.UAConfig) -> None:
+    # We will not do anything in a non-GCP cloud
+    cloud_id, _ = get_cloud_type()
+    if not cloud_id or cloud_id != "gce":
+        # If we are not running on GCP cloud, we shouldn't run this
+        # job anymore
+        cfg.gcp_auto_attach_timer = 0
```

Comment:
Since we are performing a config change/override. It's worth us logging this as a debug message. "Disabling gcp_auto_attach_timer. Cloud {} is not gce.".format(cloud_id)